zeppelin-server/src/main/java/org/apache/zeppelin/rest/NotebookRestApi.java

```diff
@@ -770,14 +770,21 @@ public Response runParagraph(@PathParam("noteId") String noteId,
       throws IOException, IllegalArgumentException {
     LOG.info("run paragraph job asynchronously {} {} {}", noteId, paragraphId, message);
 
+    Note note = notebook.getNote(noteId);
+    checkIfNoteIsNotNull(note);
+    checkIfUserCanRun(noteId, "Insufficient privileges you cannot run job for this note");
```

Comment:
I'm not sure I fully understand `you cannot run job for this note` - could we clarify?

Reply:
I brought back a log from [[ZEPPELIN-2848]](https://github.com/apache/zeppelin/pull/2526) which was deleted in [[ZEPPELIN-3681]](https://github.com/apache/zeppelin/pull/3119/files#diff-dff367ac423a9016688dc86ee27e4929R773), and didn't notice that user privileges will be checked in `NotebookService`, so It is enough to check for null value.